File: Individual/Task1MyPharmacy/Task1MyPharmacy/src/menus/MainMenu.cpp
#include <iostream>
#include "MainMenu.h"

void MainMenu::ShowMainMenu()
{
	PrintMenuForm();
	int i;
	do
	{
		i = GETCH;
		switch (i)
		{
			case MainMenuCommands::ShowMedicinesTable:
			{
				tableMenu.ShowTableMenu(1);
				PrintMenuForm();
				break;
			}
			case MainMenuCommands::ShowCosmeticsTable:
			{
				tableMenu.ShowTableMenu(2);
				PrintMenuForm();
				break;
			}
			case MainMenuCommands::Quit:
			{
				i = 0;
				std::cout << "Quit Pharmacy? | 0 : No | 1 : Yes" << std::endl;
				int j = GETCH;
				if (j == 0)
				{
					i = -1;
					PrintMenuForm();
				}
				break;
			}
			default:
			{
				continue;
			}
		}
	} while (i != MainMenuCommands::Quit);
	
	PrintPharmacyGetWellSoon();
}

void MainMenu::PrintMenuForm()
{
	std::system("cls");
	std::cout << MainMenuCommands::Quit << " : Quit | "
		<< MainMenuCommands::ShowMedicinesTable << " : Choose Medicines Table | " 
		<< MainMenuCommands::ShowCosmeticsTable << " : Choose Cosmetics Table" << std::endl;

	PrintPharmacyWellcome();
	SetState(MainMenuWaiting);
}

void MainMenu::PrintPharmacyWellcome()
{
	std::cout << "\n\n\n";
	std::cout << "________________________________________________________" << std::endl;
	std::cout << "\n\n\n";
	std::cout << "\t\t\tPharmacy +" << std::endl;
	std::cout << "\n\n\n";
	std::cout << "________________________________________________________" << std::endl;
	std::cout << "\n\n\n";
}

void MainMenu::PrintPharmacyGetWellSoon()
{
	std::system("cls");
	std::cout << "\n\n\n";
	std::cout << "________________________________________________________" << std::endl;
	std::cout << "\n\n\n";
	std::cout << "\t\t\tGetWellSoon !" << std::endl;
	std::cout << "\n\n\n";
	std::cout << "________________________________________________________" << std::endl;
	std::cout << "\n\n\n";
}

File: Individual/MyPharmacy/AdminConsole/PL/Impl/MainMenu.cs
using System;

namespace AdminConsole.PL.Impl
{
    public class MainMenu
    {
		public enum MainMenuCommands
		{
			Quit,
			ShowMedicinesTable,
			ShowCosmeticsTable,
			MainMenuWaiting,
		};
		private MainMenuCommands mState;
		private MedicineMenu m;
		private CosmeticMenu c;

		public MainMenu()
		{
			m = new MedicineMenu();
			c = new CosmeticMenu();
		}
		public void ShowMainMenu() 
		{
			PrintMenuForm();
			//char i = Console.ReadKey().KeyChar;
			MainMenuCommands command;
			do
			{
				//char ch = Console.ReadKey(true).KeyChar;
				//command = (MainMenuCommands)Convert.ToInt32(ch-48);
				//command = (MainMenuCommands)(int.Parse(Console.ReadKey(true).KeyChar));
				//int i;
				command = (MainMenuCommands)((int)(Console.ReadKey(true).KeyChar)-48);
				switch (command)
				{
					case MainMenuCommands.ShowMedicinesTable :
					{
						//m.ShowTableMenu();
						PrintMenuForm();
						break;
					}
					case MainMenuCommands.ShowCosmeticsTable:
					{
						c.ShowTableMenu();
						PrintMenuForm();
						break;
					}
					case MainMenuCommands.Quit:
					{
						Console.WriteLine("Quit Pharmacy? | 0 : No | 1 : Yes");
						//int j = GETCH;
						if (Console.ReadKey(true).KeyChar == '1')
						{
							command = MainMenuCommands.Quit;
							//Console.Clear();
							PrintMenuForm();
						}
						break;
					}
					default:
					{
						continue;
					}
				}
			} while (command != MainMenuCommands.Quit);
			Console.Clear();
			PrintPharmacyGetWellSoon();
		}
		public void PrintPharmacyWellcome() 
		{
			Console.WriteLine("\n\n\n");
			Console.WriteLine("________________________________________________________");
			Console.WriteLine("\n\n\n");
			Console.WriteLine("\t\t\tPharmacy +");
			Console.WriteLine("\n\n\n");
			Console.WriteLine("________________________________________________________");
			Console.WriteLine("\n\n\n");
		}
		public void PrintPharmacyGetWellSoon() 
		{
			Console.WriteLine("\n\n\n");
			Console.WriteLine("________________________________________________________");
			Console.WriteLine("\n\n\n");
			Console.WriteLine("\t\t\tGetWellSoon +");
			Console.WriteLine("\n\n\n");
			Console.WriteLine("________________________________________________________");
			Console.WriteLine("\n\n\n");
		}
		public void PrintMenuForm() 
		{
			Console.Clear();
			Console.Write($"{(int)MainMenuCommands.Quit } : Quit | ");
			Console.Write($"{(int)MainMenuCommands.ShowMedicinesTable} : Choose Medicines Table | ");
			Console.WriteLine($"{(int)MainMenuCommands.ShowCosmeticsTable} : Choose Cosmetics Table");
			PrintPharmacyWellcome();
			SetState(MainMenuCommands.MainMenuWaiting);
		}
		public void SetState(MainMenuCommands state)
		{
			this.mState = state;
		}
	}
}